File: src/test/java/de/pflugradts/pwman3/domain/model/transfer/InputFaker.java
package de.pflugradts.pwman3.domain.model.transfer;

import lombok.AccessLevel;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
public class InputFaker {

    private String message;

    public static InputFaker faker() {
        return new InputFaker();
    }

    public InputFaker fakeInput() {
        this.message = "hi";
        return this;
    }

    public InputFaker withMessage(final String message) {
        this.message = message;
        return this;
    }

    public Input fake() {
        return Input.of(Bytes.of(message));
    }

}

File: src/test/kotlin/de/pflugradts/passbird/domain/model/transfer/InputFaker.kt
package de.pflugradts.passbird.domain.model.transfer

import de.pflugradts.passbird.domain.model.shell.Shell.Companion.shellOf
import de.pflugradts.passbird.domain.model.transfer.Input.Companion.inputOf

fun fakeInput(withMessage: String = "foo") = inputOf(shellOf(withMessage))

fun fakeInput(withMessage: Char = 'f') = fakeInput("$withMessage")
